environ/environ.py

```diff
@@ -970,9 +970,15 @@ def _keep_escaped_format_characters(match):
             m1 = re.match(r'\A(?:export )?([A-Za-z_0-9]+)=(.*)\Z', line)
             if m1:
                 key, val = m1.group(1), m1.group(2)
-                m2 = re.match(r"\A'(.*)'\Z", val)
+                # search for value within quotes, ignore comments after # (outside quotes)
+                m2 = re.match(r"\A\s*'(?<!\\)(.*)'\s*(#.*\s*)?\Z", val)
                 if m2:
                     val = m2.group(1)
+                else:
+                    # if no quotes, search for value, ignore comments after first #
```

Comment:
I paraphrased and reformatted your comment a bit to make it clearer and satisfy the code style.

```suggestion
                    # For no quotes, find value, ignore comments
                    # after the first #
```